newton/_src/sim/builder.py

```diff
@@ -5076,6 +5335,18 @@ def finalize(self, device: Devicelike | None = None, requires_grad: bool = False
             m.body_key = self.body_key
             m.body_world = wp.array(self.body_world, dtype=wp.int32)
 
+            # body colors
+            if not self.body_color_groups and len(self.body_q) > 0:
+                # If no explicit rigid-body coloring was requested, assign all bodies to a single color group.
+                # This mirrors the default behavior of the rigid-body coloring utilities.
+                self.body_color_groups = [np.arange(len(self.body_q), dtype=int)]
```

Comment:
Can't we just leave `m.body_colors` and `m.body_color_groups` set to `None` (which should be the default anyway for Warp arrays on Model) and only allocate if actually requested i.e. once the user called `builder.color()`? The VBD solver can raise an error if it is missing this field. Otherwise this might slow down the `finalize()` step unnecessarily.